test/src/main/java/org/jvnet/hudson/test/WarExploder.java

```diff
@@ -87,8 +87,8 @@ private static File explode() throws Exception {
             throw new AssertionError("jenkins.war is not in the classpath. If you are running this from the core workspace, run 'mvn install' to create the war image in war/target/jenkins");
         File war = Which.jarFile(Class.forName("executable.Executable"));
 
-        // TODO this assumes that the CWD of the Maven process is the plugin ${basedir}, which may not be the case
```

Comment:
Do not delete this comment, which AFAIK is still a valid problem. It is also noted in `JenkinsRule` sources. `test/pom.xml` already defines `buildDirectory` to be an absolute path, it seems. But `plugins/pom.xml` leaves it unset, I think letting the fallback value be just `target` as a path relative to CWD. I presume the Gradle harness is already setting it; make sure it sets it to an absolute path.